### PR TITLE
[LIBCLOUD-740] Add Point DNS provider implementation

### PR DESCRIPTION
**Pull request by aleGpereira**

### CODE REVIEW

**Comment by aleGpereira** [comment]
I've noticed i forgot to add tests for a couple of classes. Do not merge this one yet please :)

File: libcloud/dns/drivers/pointdns.py [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+
+        :param redirect_id: Redirect id.
+        :type redirect_id: ``str``
+        """
+        response = self.connection.request('/zones/%s/redirects/%s' %
+                                           (zone.id, redirect_id))
+        redirect = self._to_redirect(response.object, zone.id)
+        return redirect
+
+    def ex_get_mailredirects(self, zone, mail_r_id):
```
Comment (Kami) [inline libcloud/dns/drivers/pointdns.py:452]: I will make a minor change while merging this pr - `mailredirects` -> `mail_redirects` hope you don't mind :)

File: libcloud/dns/drivers/pointdns.py [hunk 2]
```diff
@@ -0,0 +1,626 @@
...
+            r_json['iframe_title'] = iframe
+        if query is not None:
+            r_json['redirect_query_string'] = query
+        r_data = json.dumps({'zone_redirect': r_json})
+        response = self.connection.request('/zones/%s/redirects' % zone.id,
+                                           method='POST', data=r_data)
+        redirect = self._to_redirect(response.object, zone.id)
+        return redirect
+
+    def ex_create_mailredirect(self, destination, source, zone):
```
Comment (Kami) [inline libcloud/dns/drivers/pointdns.py:420]: In the future please also add test cases for all the extension methods. Thanks.

**Comment by Kami** [comment]
Merged, thanks!

**Pull request closed without merging**